spotlight-client/src/NarrativeLayout/NarrativeLayout.tsx

```diff
@@ -0,0 +1,190 @@
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import { navigate, useParams } from "@reach/router";
+import useBreakpoint from "@w11r/use-breakpoint";
+import { rem } from "polished";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { InView } from "react-intersection-observer";
+import { useSpring } from "react-spring/web.cjs";
+import Sticker from "react-stickyfill";
+import styled from "styled-components/macro";
+import { NAV_BAR_HEIGHT } from "../constants";
+import getUrlForResource from "../routerUtils/getUrlForResource";
+import normalizeRouteParams from "../routerUtils/normalizeRouteParams";
+import { X_PADDING } from "../SystemNarrativePage/constants";
+import NarrativeNavigation from "./NarrativeNavigation";
+import { LayoutSection } from "./types";
+
+const Wrapper = styled.article`
+  display: flex;
+`;
+
+const NavContainer = styled.div`
+  flex: 0 0 auto;
+  width: ${rem(X_PADDING)};
+`;
+
+const NavStickyContainer = styled.div`
+  display: flex;
+  height: calc(100vh - ${rem(NAV_BAR_HEIGHT)});
+  position: sticky;
+  top: ${rem(NAV_BAR_HEIGHT)};
+`;
+
+export const SectionsContainer = styled.div`
+  flex: 1 1 auto;
+  /* min-width cannot be auto or children will not shrink when viewport does */
+  min-width: 0;
+`;
+
+type NarrativeLayoutProps = {
+  sections: LayoutSection[];
+};
+
+const NarrativeLayout: React.FC<NarrativeLayoutProps> = ({ sections }) => {
+  const routeParams = useParams();
+  const sectionsContainerRef = useRef() as React.MutableRefObject<
+    HTMLDivElement
+  >;
+  const showSectionNavigation = useBreakpoint(true, ["mobile-", false]);
+
+  // automated scrolling is a special case of section visibility;
+  // this flag lets us suspend in-page navigation actions while it is in progress
+  // (it is a local variable rather than a piece of React state because
+  // the animation functions are outside the render loop and won't receive state updates)
+  let isScrolling = false;
+  const cancelAutoScroll = () => {
+    isScrolling = false;
+  };
+
+  const [activeSection, directlySetActiveSection] = useState(
+    // make sure we consume the section number in the URL, if any, on first mount
+    Number(routeParams.sectionNumber) || 1
+  );
+  // wrap the section state setter in a function that respects the flag
+  const setActiveSection = useCallback(
+    (sectionNumber: number) => {
+      if (!isScrolling) {
+        directlySetActiveSection(sectionNumber);
+      }
+    },
+    [isScrolling]
+  );
+  // keep section state in sync with URL if it changes externally (e.g. via nav link)
+  useEffect(() => {
+    directlySetActiveSection(Number(routeParams.sectionNumber) || 1);
+  }, [routeParams.sectionNumber]);
+
+  const [, setScrollSpring] = useSpring(() => ({
```

Comment:
The scrolling is much smoother now than last time I reviewed a PR. not sure if it is this work or in a previous PR but its a big improvement!

Reply:
it mostly happened as part of the responsive design PR from last week or whenever that was but yeah. the "snapping" behavior has been de-emphasized to the point where it barely happens at all (any user-initiated scroll events cancel it, so it doesn't keep you from scrolling or jump around in a janky way). might still need to be fine tuned but I'm glad you feel an improvement!